ui/ui-frontend/projects/archive-search/src/app/archive/archive-search/archive-search.component.spec.ts

```diff
@@ -385,5 +387,92 @@ describe('ArchiveSearchComponent', () => {
         }),
       );
     });
+    it('should update criteria when a virtual node is checked', async () => {
+      await setupTest({ opi: '1234' });
+
+      await fixture.whenStable();
+
+      const virtualNode1: NodeData = {
+        checked: true,
+        virtualPath: 'virtualPath',
+        id: 'virtualPath',
+        realParentId: 'someRealParentId',
+        realParentTitle: 'someRealParentTitle',
+        isVirtual: true,
+        title: 'virtualPath',
+      };
+
+      const virtualNode2: NodeData = {
+        checked: true,
+        virtualPath: 'virtualPath',
+        id: 'virtualPath',
+        realParentId: 'someOtherRealParentId',
+        realParentTitle: 'someOtherRealParentTitle',
+        isVirtual: true,
+        title: 'virtualPath',
+      };
+
+      archiveSharedDataService.emitNode(virtualNode1);
+      archiveSharedDataService.emitNode(virtualNode2);
+      fixture.detectChanges();
+
+      expect(component.searchCriterias.has('VIRTUAL')).toBeTrue();
+
+      const virtualCriteria = component.searchCriterias.get('VIRTUAL');
+      expect(virtualCriteria?.values?.length).toEqual(2);
+
+      const virtualValues = virtualCriteria.values.filter((value) => value?.value?.id === 'virtualPath');
+
+      expect(['someRealParentId', 'someOtherRealParentId']).toContain(virtualValues[0].value.virtualNodeRealParentId);
+      expect(['someRealParentId', 'someOtherRealParentId']).toContain(virtualValues[1].value.virtualNodeRealParentId);
```

Comment:
Peut être remplacé par
```suggestion
       expect(virtualValues.map((o) => o.value.virtualNodeRealParentId)).toEqual(
        arrayWithExactContents(['someRealParentId', 'someOtherRealParentId']),
      );
```

Reply:
Ok,